Infrastructure env creation (#6)
ldclakmal/ballerina-performance-aws-ecs: README.md

# Ballerina Performance Tests on Amazon ECS

> Ballerina performance artifacts are used to continuously test the performance of Ballerina services.

Reviewer: ```suggestion
Ballerina performance artifacts are used to continuously test the performance of Ballerina language versions.
```

These performance test scripts make use of Apache JMeter and a simple Netty Backend Service, which can echo back any 
requests and also add a configurable delay to the response.

> In order to support a large number of concurrent users, two or more JMeter Servers can be used.

Reviewer: Let's remove this as well.
```suggestion
```

> Using Amazon Cloudformation, an EC2(Amazon Elastic Compute Cloud) stack which acts as a local machine and an ECS (Amazon Elastic Container Service) stack compromising of a Netty server and Ballerina service are created.

Reviewer: ```suggestion
Use AWS Cloudformation, AWS EC2 (Amazon Elastic Compute Cloud) which acts as a local machine and AWS ECS (Amazon Elastic Container Service) for Netty echo server and Ballerina service.
```

To fully automate the performance tests on amazon ECS, an AWS CloudFormation template is used to create a deployment of an EC2 
instance and an ECS cluster compromising a Ballerina and a Netty Backend Service.

## About Ballerina

Ballerina makes it easy to write microservices that integrate APIs.

#### Integration Syntax
A compiled, transactional, statically and strongly typed programming language with textual and graphical syntaxes. Ballerina incorporates fundamental concepts of distributed system integration and offers a type safe, concurrent environment to implement microservices.

#### Networked Type System
A type system that embraces network payload variability with primitive, object, union, and tuple types.

#### Concurrency

> An execution model composed of lightweight parallel worker units that are non-blocking where no function can lock an executing thread manifesting sequence concurrency.

Reviewer: Let's remove this part.

## Run Ballerina Performance Tests on Amazon ECS

You can run Ballerina Performance Tests from the source using the following instructions.

### Prerequisites

> * [AWS CLI](https://aws.amazon.com/cli/) - Please make sure to [configure the AWS Cli](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html)

Reviewer: ```suggestion
* [AWS CLI](https://aws.amazon.com/cli/) - Please make sure to [configure the AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html)
```
